eslint.config.js

```diff
@@ -0,0 +1,22 @@
+const ts = require('typescript-eslint');
+const js = require('@eslint/js');
+const prettier = require('eslint-config-prettier');
+
+module.exports = [
+  js.configs.recommended,
+  ts.configs.eslintRecommended,
+  prettier,
+  {
+    ignores: ['lib/**'],
+  },
+  {
+    languageOptions: {
+      globals: {
+        console: true,
+        exports: true,
+        module: true,
+        require: true,
+      },
```

Comment:
Note: Alternatively, we could also use https://github.com/sindresorhus/globals but I wasn't too keen on that dependency just for this. Actually wish eslint came out of the box with support for the most common environments (browser & node), with the option of plugging in others.

Reply:
I think globals are configured when an `environment` is defined. 

So we could leave this out and let the concrete projects define the env and thus its globals.

Reply:
I don't believe `environment` exists anymore in the [new config](https://eslint.org/docs/latest/use/configure/configuration-files-new). All is managed via `globals` now, it seems.
But of course, the concrete projects can define their own globals, or use dependencies that define globals for them.